From bc54390e2af7493bc356cb8ae820c4fd485a09b6 Mon Sep 17 00:00:00 2001
Message-Id: <bc54390e2af7493bc356cb8ae820c4fd485a09b6.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Akash Goel <akash.goel@intel.com>
Date: Tue, 10 Jun 2014 16:07:43 +0530
Subject: [PATCH 112/312] FOR_UPSTREAM [VPG]: drm/i915: Fix enable of proper
 Interrupts based on the turbo selection

Enable only the required interrupt(GEN6_PM_RP_UP_EI_EXPIRED) for VLV
C0 residency based turbo workaround.

Issue: VIZ-3935
Change-Id: I99c61d3e7886158d227eebd7058daa641ab68be3
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Akash Goel <akash.goel@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index af70916..dad6857 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -3450,6 +3450,7 @@ static void gen5_gt_irq_postinstall(struct drm_device *dev)
 
 		if (dev_priv->rps.use_RC0_residency_for_turbo) {
 			dev_priv->pm_irq_mask &= ~GEN6_PM_RP_UP_EI_EXPIRED;
+			pm_irqs &= ~GEN6_PM_RPS_EVENTS;
 			pm_irqs |= GEN6_PM_RP_UP_EI_EXPIRED;
 		} else {
 			dev_priv->pm_irq_mask &= ~GEN6_PM_RPS_EVENTS;
-- 
1.7.9.5

